### PR TITLE
Swarmers can no longer breach the station.

### LINKED ISSUE #21075: Swarmers can attack and destroy airlocks leading to space
@phil235 
I don't think swarmers had any restrictions on attacking airlocks before, since they couldn't destroy them.

### PR DESCRIPTION
**Pull request by F-OS**
i am new to coding and it is 4 in the morning so this might not work.

fixes #21075

:cl: F-OS
fix: swarmers can no longer destroy airlocks.
/:cl:

### CODE REVIEW

**Comment by F-OS** [comment]
Testing if this works fully right now, DNM.

**Comment by F-OS** [comment]
100% tested, ready for merging.

**Comment by optimumtact** [comment]
You didn't justify your pr and you were snarky about it

**Comment by ChangelingRain** [comment]
Fixes shouldn't require justification

**Comment by F-OS** [comment]
fixed it for optimumtact.

**Comment by optimumtact** [comment]
I actually assumed it was a feature, that's my bad

**Comment by F-OS** [comment]
this is slightly a feature because swarmers now deconstruct airlocks like other machines, but is mainly a bugfix to stop swarmers from breaching the station.

**Comment by PKPenguin321** [comment]
?????? Then you've failed since they can still breach the station by just eating walls?

**Comment by F-OS** [comment]
uhm no they cannot @PKPenguin321 refer to this block of code please https://github.com/F-OS/tgstation/blob/a56a4dd7cdfd908328577e5c6935f288db2ed191/code/game/gamemodes/miniantags/bot_swarm/swarmer.dm#L329

**Comment by PKPenguin321** [comment]
OH, you mean like breaching as in causing hull breaches.

You should have made that clearer.

**Comment by F-OS** [comment]
yeah i shoulda made that clearer sorry.

**Comment by KorPhaeron** [comment]
Make breaking airlocks use the same check that they get when trying to break walls or windows, don't cut out airlock smashing entirely.

**Comment by F-OS** [comment]
got it kor. one sec.

**Comment by Supermichael777** [comment]
why not generalize this to all structures?

**Comment by F-OS** [comment]
sorry it took me a while to respond, the reason i am not generalizing it is because i am bad at coding.

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 1]
```diff
@@ -242,8 +242,12 @@
 	return TRUE
 
 /obj/machinery/door/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	for(var/turf/T in range(1, src))
+		if(isspaceturf(T) || istype(T.loc, /area/space))
+			S << "<span class='warning'>Destroying this object has the potential to cause a hull breach. Aborting.</span>"
+			S.target = null
+			return TRUE
 	S.DisIntegrate(src)
-	return TRUE
```
Comment (Shadowlight213) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm]: No. This should be returning true so AI swarmers see the airlock as a valid object if they succeed in eating it.

**Comment by F-OS** [comment]
just want to say this pr is 100 percent ready for a testmerge.

**Comment by F-OS** [comment]
woot! second merged pr!

**Pull request merged**